types/index.d.ts

```diff
@@ -0,0 +1,369 @@
+    readonly create: Schema<Registry>["create"];
+
+    /** Whether or not Mirage should log all requests/response cycles. */
+    logging: boolean;
+
+    /** A default number of ms to artifically delay responses for all routes. */
+    timing: number;
+
+    /** A default prefix applied to all subsequent route definitions. */
+    namespace: string;
+
+    /** Creates multiple models of the given type. */
+    createList<
+      K extends keyof Registry,
+      Init extends Registry[K],
+      Data extends Partial<Init>
+    >(modelName: K, count: number, data?: Data): Array<Init & Data>;
+
+    /** Handle a GET request to the given path. */
+    get(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Handle a POST request to the given path. */
+    post(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Handle a PUT request to the given path. */
+    put(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Handle a PATCH request to the given path. */
+    patch(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Handle a DELETE request to the given path. */
+    del(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Pass through one or more URLs to make real requests. */
+    passthrough(...urls: string[]): void;
+
+    /** Load all available fixture data matching the given name(s). */
+    loadFixtures(...names: string[]): void;
+
+    /** Shutdown the server and stop intercepting network requests. */
+    shutdown(): void;
+  }
+}
+
+declare module "miragejs/db" {
+  /** The in-memory database containing all currently active data keyed by collection name. */
+  export default class Db {
+    [key: string]: unknown;
+  }
+}
+
+declare module "miragejs/orm/schema" {
+  import { Collection, ModelInstance } from "miragejs";
+  import Db from "miragejs/db";
+
+  type ModelInitializer<Data> = {
+    [K in keyof Data]: Data[K] extends Collection<infer M>
+      ? Collection<M> | M[]
+      : Data[K];
+  };
+
+  /**
+   * An interface to the Mirage ORM that allows for querying and creating records.
+   */
+  export default class Schema<Registry = Record<string, ModelInstance>> {
+    /** Mirage's in-memory database */
+    readonly db: Db;
+
+    /**
+     * Creates a model of the given type.
+     * @param modelName The type of model to instantiate
+     * @param data Optional initial values for model attributes/relationships
+     */
+    create<
+      K extends keyof Registry,
+      Init extends Registry[K],
+      Data extends Partial<ModelInitializer<Init>>
+    >(
+      modelName: K,
+      data?: Data
+    ): Init & { [K in keyof Init & keyof Data]: Exclude<Init[K], undefined> };
+
```

Comment:
`update` method missing

Reply:
I don't see any `update` method in the `Schema` documentation .
https://miragejs.com/api/classes/schema

Reply:
Oh, you are right. I was misleading this with `schema.db.todos.update` for instance

Reply:
No worries. Yep, it comes from the `model`, so it should work... ;)